staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -593,9 +593,11 @@ func (s *store) GetList(ctx context.Context, key string, opts storage.ListOption
 
 	// set the appropriate clientv3 options to filter the returned data set
 	var limitOption *clientv3.OpOption
-	var limit int64 = pred.Limit
+	limit := pred.Limit
+	var paging bool
```

Comment:
I'm not opposed to this change, but I don't understand what do we get from it.

paging seems to be true iff s.pagingEnabled and limit>0, which is the same condition as limitOption!=nil.

What do this get us?

Reply:
This is what the code used to look like before the dynamic page size PR went in. The change to use `limitOption != nil` to mean `paging` is, to my eyes, way too complex. It is true that `limitOption` changes over time (that's what the dynamic page PR did), but `paging` the boolean was set once and read forever, it is a direct reflection of client intent.

To my eyes, `paging` was a simple thing to understand. Now, with the changes to `limiOption != nil` (which I will say were also not strictly necessary to that pull...) I have to understand the semantic of `limitOption`, how it changes over time, when it is and is not `nil`... and after doing all of that careful reading you notice that `limitOpion != nil` always evaluates the same way as the boolean used to, which is much easier to read and understand.

Reply:
I would say it's a matter of personal preference - both are technically correct.
I guess my bias may be coming from being familiar with this code. If it seems visibly more difficult to follow without this PR, I'm not going to oppose (even though it's not what I would personally do).

Reply:
I guess `paging` is a static `boolean` and you can look up the value quickly and easily, and always be certain you understand what it means. With the `nil` check you need to read the flow of the method and be certain that you understand all of the cases where it might and might not be `nil`, while understanding how it is updated, etc ...